src/js/components/search/SearchWidget.vue

```diff
@@ -1,14 +1,28 @@
 <template>
-    <div class="search-container">
-        <input type="text" class="search-input" v-model="searchText" placeholder="Search by name..." />
-        <ResultsList v-if="!isLoading" :artifacts="filteredArtifacts" :heroes="filteredHeroes" />
-    </div>
+    <section class="section-container">
+        <LoadingMessage :isLoading="isLoading" />
```

Comment:
Don't you think the LoadingMessage would look better if it only showed up if the user was already typing a search when the promise wasn't fulfilled?  I was originally thinking that ResultsList would have the v-if="!isLoading" and the LoadingMessage would appear beneath the input if isLoading and trimmedSearch.

Reply:
Well, the xhr call already happens the moment the `mounted()` is called (you can see the brown bar loading indicator at the top of the page), so I just added the same loader to the section.

It would only make sense to have a loading when typing if the calls were not triggered on mounted, but on the first model change of `searchText`.

That could be an improvement (having the xhr only be called if input was interacted with).